### PR TITLE
Simplify build system.

### PR DESCRIPTION
**Pull request by jdek**
CMake is definitely unneeded, it's massive, and yet another dependency. I've added a fairly simple makefile to replace it, and the install/uninstall scripts. I've also reverted the commits which vendor jack into this repo and patch it--this is very bad practise and shouldn't be in here.

p.s. looks like a cool project, I was contemplating writing a JACK driver for OSX to replace the old JACK driver which has long bitrot; but now I've found this, I'd be happy to help.

### CODE REVIEW

File: CaptainJack.driver/Contents/MacOS/CaptainJack [hunk 1]
```diff
@@ -1 +0,0 @@
-../../../build/src/captain-jack
```
Comment (Qix-) [inline CaptainJack.driver/Contents/MacOS/CaptainJack:1]: So this is necessary, though in your version it's just `../../../build/captain-jack`. Your `stage` step performs a build-in-source kind of environment which I try to stay away from.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+all: $(BUILDDIR)/captain-jack-daemon $(BUILDDIR)/captain-jack
+
+.PHONY: clean
+clean:
+	rm -rf $(BUILDDIR)
+
+# Install
+.PHONY: stage
+stage: $(BUILDDIR)/captain-jack
+	install -m 0755 $(BUILDDIR)/captain-jack $(DRIVER_NAME)/Contents/MacOS/
```
Comment (Qix-) [inline Makefile]: See comment above :)

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+.PHONY: install
+uninstall: stop
+	sudo rm -vf $(DESTDIR)$(PREFIX)/sbin/captain-jack-daemon
+	sudo rm -vf $(DESTDIR)/Library/LaunchDaemons/me.junon.CaptainJack.plist
+	sudo rm -vrf $(DESTDIR)$(PLUGINDIR)/CaptainJack.driver
+
+.PHONY: kill_audio
+kill_audio:
+	ps aux | grep _coreaudiod | grep -v grep | awk '{print $2}' | xargs sudo kill -9
```
Comment (Qix-) [inline Makefile]: needs to be `print $$2` or else it just evaluates to `awk '{print }'` and blows up on uninstall, haha.

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+	install -m 0755 $(BUILDDIR)/captain-jack $(DRIVER_NAME)/Contents/MacOS/
+
+.PHONY: install
+install: $(BUILDDIR)/captain-jack-daemon stage
+	sudo install -m 0755 -o root -g wheel $(BUILDDIR)/captain-jack-daemon $(DESTDIR)$(PREFIX)/sbin
+	sudo install -m 0644 -o root -g wheel $(DRIVER_NAME)/Contents/Info.plist $(DESTDIR)$(PLUGINDIR)/$(DRIVER_NAME)/Contents/Info.plist
+	sudo install -m 0755 -o root -g wheel $(DRIVER_NAME)/Contents/MacOS/CaptainJack $(DESTDIR)$(PLUGINDIR)/$(DRIVER_NAME)/Contents/MacOS/CaptainJack
+	sudo install -m 0644 -o root -g wheel $(DRIVER_NAME)/Contents/Resources/DeviceIcon.icns $(DESTDIR)$(PLUGINDIR)/$(DRIVER_NAME)/Contents/Resources/DeviceIcon.icns
+	sudo install -m 0644 -o root -g wheel $(DRIVER_NAME)/Contents/Resources/English.lproj/Localizable.strings $(DESTDIR)$(PLUGINDIR)/$(DRIVER_NAME)/Contents/Resources/English.lproj/Localizable.strings
+	sudo install -m 0600 -o root -g wheel me.junon.CaptainJack.plist $(DESTDIR)/Library/LaunchDaemons/
```
Comment (Qix-) [inline Makefile]: If the destination directory doesn't exist, this fails. Apparently `install -D` on other platforms works but OS/X doesn't seem to have that flag. Looks like it'll have to be a `mkdir -p` call.

File: CaptainJack.driver/Contents/MacOS/CaptainJack (hunk 1 above)
Comment (Qix-) [inline CaptainJack.driver/Contents/MacOS/CaptainJack:1]: Nevermind, I see what happened. `install` doesn't like symlinks. Hmm.. okay. Disregard this.

File: Makefile (hunk 4 above)
Comment (jdek) [reply Makefile]: osx has `install -d`, I'll add that

File: Makefile (hunk 3 above)
Comment (jdek) [reply Makefile]: Ah! thanks

**Comment by jdek** [comment]
I've integrated your comments, and updated the PR.

**Comment by Qix-** [comment]
Here is the diff I got to, on top of this PR.

```
diff --git a/Makefile b/Makefile
index 03e0c40..3cf425a 100644
--- a/Makefile
+++ b/Makefile
@@ -12,7 +12,7 @@ CFLAGS_CJ  = -bundle -framework CoreAudio -framework CoreFoundation
 CFLAGS_CJD = -framework CoreFoundation
 DESTDIR     =
 PREFIX      = /usr/local
-PLUGINDIR  = /Library/Audio/Plug-Ins/HAL/
+PLUGINDIR  = /Library/Audio/Plug-Ins/HAL
 BUILDDIR    = build

 SRCS        = $(wildcard **/*.c)
@@ -52,10 +52,14 @@ clean:
 # Install
 .PHONY: stage
 stage: $(BUILDDIR)/captain-jack
-   install -m 0755 $(BUILDDIR)/captain-jack $(DRIVER_NAME)/Contents/MacOS/
+   mkdir -p $(DRIVER_NAME)/Contents/MacOS
+   install -m 0755 $(BUILDDIR)/captain-jack $(DRIVER_NAME)/Contents/MacOS/CaptainJack

 .PHONY: install
 install: $(BUILDDIR)/captain-jack-daemon stage
+   sudo mkdir -p $(DESTDIR)$(PLUGINDIR)/$(DRIVER_NAME)/Contents/MacOS
+   sudo mkdir -p $(DESTDIR)$(PLUGINDIR)/$(DRIVER_NAME)/Contents/Resources
+   sudo mkdir -p $(DESTDIR)$(PLUGINDIR)/$(DRIVER_NAME)/Contents/Resources/English.lproj
    sudo install -m 0755 -o root -g wheel $(BUILDDIR)/captain-jack-daemon $(DESTDIR)$(PREFIX)/sbin
    sudo install -m 0644 -o root -g wheel $(DRIVER_NAME)/Contents/Info.plist $(DESTDIR)$(PLUGINDIR)/$(DRIVER_NAME)/Contents/Info.plist
    sudo install -m 0755 -o root -g wheel $(DRIVER_NAME)/Contents/MacOS/CaptainJack $(DESTDIR)$(PLUGINDIR)/$(DRIVER_NAME)/Contents/MacOS/CaptainJack
@@ -71,7 +75,7 @@ uninstall: stop

 .PHONY: kill_audio
 kill_audio:
-   ps aux | grep _coreaudiod | grep -v grep | awk '{print $2}' | xargs sudo kill -9
+   ps aux | grep _coreaudiod | grep -v grep | awk '{print $$2}' | xargs sudo kill -9

 .PHONY: stop
 stop: kill_audio
```

File: Makefile (hunk 4 above)
Comment (Qix-) [reply Makefile]: `install -d` isn't the same though. I'm not sure what it does but it outputs the following:

```
sudo install -d -m 0755 -o root -g wheel build/captain-jack-daemon /usr/local/sbin
install: build/captain-jack-daemon exists but is not a directory
make: *** [install] Error 71
```

It looks as if it's meant to install an entire directory and expects its operands to be directories. Completely misleading `man` page :/

**Comment by jdek** [comment]
Ok. Take two.

**Comment by Qix-** [comment]
Perfect, there we go :) I'm okay with this. I normally prefer CMake because I like [Ninja](https://ninja-build.org/), but I'm okay to do without it for now.

I also don't mind building against the system `jack` installation, even though that [doesn't seem to work well](https://github.com/jackaudio/jack1/issues/41) right now.

**Comment by jdek** [comment]
If the project becomes a lot more complex (with many more components), then it might make sense to switch back to CMake,

Unrelated (didn't think it warranted a separate issue): Could you add a README stating how to actually use this?

**Comment by Qix-** [comment]
Yeah I can; It's still under development so the next phase is to actually wire it up to JACK.

Give me a few minutes.

**Comment by Qix-** [comment]
@joshdekock Added an initial readme. As I said it's still under development.

**Pull request merged**